### ISSUE TITLE
Define and Document deployment and release processes

### ISSUE THREAD

**Issue by lkngtn** [1Hive/devops#1]
We can release redemptions as part of this 🚀

**Comment by yeqbfgxjiq** [comment]
### A1
- [`aragon/deployments`](https://github.com/aragon/deployments) - documents both the contract deployments as well as IPFS content that is served to users on production via [aragonPM](https://hack.aragon.org/docs/cli-apm-commands)
- [`aragon/aragon/deployments`](https://github.com/aragon/aragon/deployments) - used only for developer previews and ease of testing pull requests and is served via [Now for GitHub](https://zeit.co/github)
- [AragonPM Subgraph](https://thegraph.com/explorer/subgraph/aragonblack/aragonpm) - A subgraph for AragonPM ([repo](https://github.com/AragonBlack/aragonpm-subgraph))

### Autark
- [TPS deployments](https://github.com/AutarkLabs/planning-suite/tree/dev/shared/deployments) - same process as A1

### Black
- TBD

### 1HIve
- [`1hive/deployments`](https://github.com/1Hive/deployments) - same process as A1

**Comment by yeqbfgxjiq** [comment]
### TL;DR: 
We could use [eth-bytecode-verifier](https://www.npmjs.com/package/eth-bytecode-verifier) to go a step beyond the A1 deployment process to make it easy for people to check the code in our [deployment repo](https://github.com/1Hive/deployments) against the deployed contracts on Ethereum.

### EDIT
It looks like `bytecode-verifier` is [no longer maintained](https://github.com/ConsenSys/bytecode-verifier/issues/12). (Sigh...) This makes me sad :/ 

<br> 

Been thinking a lot about how the security auditing process is going to look, not just from the standpoint of handing off our apps, but from the standpoint of an external project who hears about Dandelion Orgs and is considering using the DAO template. How would they know that what they're deploying and use is in fact the audited and secure code that we claim that it is?

Most projects build a thing. Some projects get that thing audited. Even fewer have ongoing audits and security best practices built into their development and release process. After doing my own research and talking to the ToB team, it's becoming clear that actually most people don't understand (or care about) Ethereum security. They simply want a stamp of assurance so that they can feel good about things. This is more on the trusting side than the verifying side. (Most) people  don't trust Ethereum contracts or DAOs because they're new and weird and the PR around them is all hype or all snafus. 

It doesn't have to be this way though. We're doing our best to make sure that our apps are easy to understand, easy to use, and secure. This way as many people as possible can empower their projects with Dandelion Orgs and discover the power of Aragon. Also hopefully this will contribute to best practices in the Aragon ecosystem. As part of our release process we should point to the deployed contracts, but also shows the source code of that contract as well as instructions to verify that the compiled source code matches the deployed contract. Consensys Dilligence has a few thoughts on this [here](https://media.consensys.net/audit-the-deployed-smart-contract-not-github-16082b2fcb1b) and they released (or funded development for) a tool called [eth-bytecode-verifier](https://www.npmjs.com/package/eth-bytecode-verifier) that allows you to locally compile source code and then verify byte code of that source code to a deployed contract.

**Comment by yeqbfgxjiq** [comment]
Now that I'm thinking about this more... I have many questions. Mainly,how do upgrades happen? With the Autark apps I've reported bugs and/or feature requests and then they've been "hot" upgraded in the DAO without me doing anything. This implies that Autark actually controls the apps, and therefor their control is centralized. This kind of goes against the whole naming convention of "decentralized" autonomous organizations. 

Would like to learn about how that process works within the Aragon stack, and then ensure that with our apps there is the option, but not the requirement, to *opt in* to automatic upgrades. This way if people want to launch a Dandelion Org that is self sovereign they can do so without relying on the 1Hive team :)

**Comment by shayanb** [comment]
> It looks like `bytecode-verifier` is [no longer maintained](https://github.com/ConsenSys/bytecode-verifier/issues/12). (Sigh...) This makes me sad :/

Updates coming up. [bytecode-verifier/tree/0.3.1](https://github.com/ConsenSys/bytecode-verifier/tree/0.3.1) is working now but needs more testing. 

Don't be sad :)

**Comment by yeqbfgxjiq** [comment]
@shayanb Thanks! :)
